src/common/services/ArtemisService.ts

```diff
@@ -0,0 +1,119 @@
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import fetch, { RequestInit } from "node-fetch";
+import { COPILOT_UNAVAILABLE } from "../copilot/constants";
+import { ITelemetry } from "../../client/telemetry/ITelemetry";
+import { sendTelemetryEvent } from "../copilot/telemetry/copilotTelemetry";
+import { CopilotArtemisFailureEvent, CopilotArtemisSuccessEvent } from "../copilot/telemetry/telemetryConstants";
+import { getCrossGeoCopilotDataMovementEnabledFlag } from "./BAPService";
+import { BAPServiceStamp as BAPAPIEndpointStamp } from "./Constants";
+import { IArtemisAPIOrgResponse, IArtemisServiceEndpointInformation, IIntelligenceAPIEndpointInformation } from "./Interfaces";
+import { isCopilotDisabledInGeo, isCopilotSupportedInGeo } from "../copilot/utils/copilotUtil";
+
+export async function getIntelligenceEndpoint(orgId: string, telemetry: ITelemetry, sessionID: string, environmentId: string): Promise<IIntelligenceAPIEndpointInformation> {
+
+    const artemisResponses = await fetchArtemisResponse(orgId, telemetry, sessionID);
+
+    if (artemisResponses === null || artemisResponses.length === 0) {
+        return { intelligenceEndpoint: null, geoName: null, crossGeoDataMovementEnabledPPACFlag: false };
+    }
+
+    const artemisResponse = artemisResponses[0];
+    if (artemisResponse !== null) {
+        const { geoName, environment, clusterNumber } = artemisResponse.response as unknown as IArtemisAPIOrgResponse;
+        sendTelemetryEvent(telemetry, { eventName: CopilotArtemisSuccessEvent, copilotSessionId: sessionID, geoName: String(geoName), orgId: orgId });
+
+        const crossGeoDataMovementEnabledPPACFlag = await getCrossGeoCopilotDataMovementEnabledFlag(artemisResponse.stamp, telemetry, environmentId);
+
+        if (isCopilotDisabledInGeo().includes(geoName)) {
+            return { intelligenceEndpoint: COPILOT_UNAVAILABLE, geoName: geoName, crossGeoDataMovementEnabledPPACFlag: crossGeoDataMovementEnabledPPACFlag };
+        }
+        else if (crossGeoDataMovementEnabledPPACFlag === true) {
+            // Do nothing - we can make this call cross geo
+        }
+        else if (!isCopilotSupportedInGeo().includes(geoName)) {
+            return { intelligenceEndpoint: COPILOT_UNAVAILABLE, geoName: geoName, crossGeoDataMovementEnabledPPACFlag: crossGeoDataMovementEnabledPPACFlag };
+        }
+
+        const intelligenceEndpoint = `https://aibuildertextapiservice.${geoName}-${'il' + clusterNumber}.gateway.${environment}.island.powerapps.com/v1.0/${orgId}/appintelligence/chat`
+
+        return { intelligenceEndpoint: intelligenceEndpoint, geoName: geoName, crossGeoDataMovementEnabledPPACFlag: false };
+    }
+
+    return { intelligenceEndpoint: null, geoName: null, crossGeoDataMovementEnabledPPACFlag: false };
+}
+
+// Function to fetch Artemis response
+export async function fetchArtemisResponse(orgId: string, telemetry: ITelemetry, sessionID = '') {
```

Comment:
that is correct - atleast from Web extension flow i see this is being called twice. Will take this up with next PR since this change is existing. In any case i have to raise a PR for desktop envId passing for this change to be complete from both web and desktop